conformance/reporter.go

```diff
@@ -0,0 +1,290 @@
+package conformance
+
+import (
+	"bytes"
+	"fmt"
+	"html/template"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+
+	"github.com/onsi/ginkgo/config"
+	"github.com/onsi/ginkgo/types"
+)
+
+const (
+	htmlTemplate string = `{{define "report"}}
+<html>
+	<head>
+		<title>OCI Distribution Conformance Tests</title>
+		<style>
+			.result {
+				padding: 1.25em 0 .25em 2em;
+				border: 2px solid #e2e2e2;
+				border-radius: 5px;
+				margin-top: 10px;
+			}
+
+			.red {
+				background: #ffc8c8;
+			}
+
+			pre.fail-message {
+				background: #f9a5a5;
+				padding: 20px;
+				display: inline-block;
+				border-radius: 4px;
+				font-size: 1.25em;
+			}
+
+			.green {
+				background: #c8ffc8;
+				padding: 1.25em 0 1.25em 2em;
+			}
+
+			.toggle {
+				border: 2px solid #3e3e3e;
+				cursor: pointer;
+				width: 1em;
+				text-align: center;
+				font-weight: bold;
+				display: inline;
+				font-family: monospace;
+				padding: 0 .25em 0 .25em;
+				margin: 1em 1em 1em 0;
+				font-size: 12pt;
+				color: #3e3e3e;
+				border-radius: 3px;
+			}
+
+			pre.pre-box {
+				background: white;
+				padding: 10px;
+				border: 1px solid gray;
+				display: inline-block;
+				border-radius: 4px;
+				width: 97%;
+				font-size: 1.25em;
+				overflow-x: auto;
+				max-height: 60em;
+				overflow-y: auto;
+			}
+		</style>
+		<script>
+			function toggleOutput(id) {
+				var elem = document.getElementById(id);
+				var button = document.getElementById(id + "-button");
+				if (elem.style['display'] === 'block') {
+					button.innerHTML = "+";
+					elem.style['display'] = 'none';
+				} else {
+					button.innerHTML = "-";
+					elem.style['display'] = 'block';
+				}
+			}
+		</script>
+	</head>
+	<body style="padding: 30px;">
+		<h1>OCI Distribution Conformance Tests</h1>
+		<div>
+			{{with .SpecSummaryMap}}
+				{{$x := .M}}
+				{{range $i, $k := .Keys}}
+					<h2>{{$k}}</h2>
+					{{$v := index $x $k}}
+					{{range $z, $s := $v}}
+						{{if eq $s.State 4}}
+						<div class="result red">
+							<div id="output-box-{{$s.ID}}-button" class="toggle"
+								onclick="javascript:toggleOutput('output-box-{{$s.ID}}')">+</div>
+							<h3 style="display: inline;">{{$s.Title}}</h3><br>
+							<div>
+								<div id="output-box-{{$s.ID}}" style="display: none;">
+									<pre class="pre-box">{{$s.CapturedOutput}}</pre>
+								</div>
+								</div>
+									<pre class="fail-message">{{$s.Failure.Message}}</pre><br>
+								</div>
+							</div>
+						{{else if eq $s.State 3}}
+						<div class="result green">
+							<div id="output-box-{{$s.ID}}-button" class="toggle"
+								onclick="javascript:toggleOutput('output-box-{{$s.ID}}')">+</div>
+							<h3 style="display: inline;">{{$s.Title}}</h3><br>
+							<div id="output-box-{{$s.ID}}" style="display: none;">
+								<pre class="pre-box">{{$s.CapturedOutput}}</pre>
+							</div>
+						</div>
```

Comment:
nit: the div closures appear to be misaligned in several spots

Reply:
fixed